### DISCUSSION TITLE
Add support for momentjs's Start of Time and End of Time.

### DISCUSSION THREAD

**Discussion by jenningsb2** [chhoumann/quickadd#650] [Ideas]
I want to append a task to my weekly note. I am unable to do this with my existing date format: `YYYY-MM-DD [Week]`

I was hoping that the QuickAdd plugin would support offsetting the current date to the start of the week.

In an ideal world, I'd be able to reference my weekly note by writing something like: `{{DATE:startOfweek [Week]}}` (or something similar)

https://momentjs.com/docs/#/manipulating/start-of/

```
moment().startOf('year');    // set to January 1st, 12:00 am this year
moment().startOf('month');   // set to the first of this month, 12:00 am
moment().startOf('quarter');  // set to the beginning of the current quarter, 1st day of months, 12:00 am
moment().startOf('week');    // set to the first day of this week, 12:00 am
moment().startOf('isoWeek'); // set to the first day of this week according to ISO 8601, 12:00 am
moment().startOf('day');     // set to 12:00 am today
moment().startOf('date');     // set to 12:00 am today
moment().startOf('hour');    // set to now, but with 0 mins, 0 secs, and 0 ms
moment().startOf('minute');  // set to now, but with 0 seconds and 0 milliseconds
moment().startOf('second');  // same as moment().milliseconds(0);
```

**Comment by johnsturgeon** [comment]
Is there any workaround?  I'd love to be able to add a task to my 'weekly' or 'monthly' notes

**Comment by chhoumann** [comment]
This shipped in QuickAdd 2.14.0 as date snapping: `|startof:<unit>` and `|endof:<unit>` on both `{{DATE}}` and `{{VDATE}}`.

The original ask works directly now:

```
{{DATE:YYYY-MM-DD [Week]|startof:week}}
```

resolves to the start-of-week date in your format (use `|startof:isoweek` if your weeks are Monday-anchored). Units: day, week, isoweek, month, quarter, year - and the `+N` day offset applies before the snap, so `{{DATE:YYYY-MM-DD+7|startof:week}}` is the start of next week.

@johnsturgeon that's the workaround-free answer for periodic notes too: point your Capture choice's target file name at the note using a snapped date, e.g. `{{DATE:YYYY-MM-DD [Week]|startof:week}}` for weeklies or `{{DATE:YYYY-MM|startof:month}}` for monthlies, and the task lands in the right note.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#date-snap

Closing as resolved.
